api/s2n.h

```diff
@@ -249,9 +249,9 @@ typedef enum {
 /**
  * Determines whether s2n-tls is operating in FIPS mode.
  *
- * s2n-tls enters FIPS mode on initialization when the linked libcrypto has FIPS mode enabled. Some
- * libcryptos, such as AWS-LC-FIPS, have FIPS mode enabled by default. With other libcryptos, such
- * as OpenSSL, FIPS mode must be enabled before initialization by calling `FIPS_mode_set()`.
+ * s2n-tls enters FIPS mode on initialization when the linked libcrypto has FIPS mode enabled.
+ * s2n-tls only supports FIPS mode when built with AWS-LC-FIPS or AWS-LC-FIPS-2022. AWS-LC-FIPS
+ * and AWS-LC-FIPS-2022 have FIPS mode enabled by default.
```

Comment:
We do still support openssl-1.0.2-fips. Arguable if that's still fips, but it does have fips mode.